Ofer/326040 list capabilities for physical storage

Oferlis opened the pull request:
Added textual summary function to display capabilities of a physical storage.
related to PR in AutoSDE provider repo  https://github.com/ManageIQ/manageiq-providers-autosde/pull/198

![Screenshot 2023-03-28 at 9 32 37](https://user-images.githubusercontent.com/62609377/228148508-af9925c3-ddd7-4c69-99c9-f870a52a0123.png)

jeffibm commented:
@Oferlis , could you please rebase with master (also squash 6 commits) branch to see if the the spec:javascript is still failing in GitHub actions

Oferlis commented:
> @Oferlis , could you please rebase with master (also squash 6 commits) branch to see if the the spec:javascript is still failing in GitHub actions

Hey, I have rebased the branch and fixed one of the failing specs, but the other keeps failing.

jeffibm commented:
> > @Oferlis , could you please rebase with master (also squash 6 commits) branch to see if the the spec:javascript is still failing in GitHub actions
> 
> Hey, I have rebased the branch and fixed one of the failing specs, but the other keeps failing.

Could you try running them locally to see if this is related to your changes?

Oferlis commented:
> > > @Oferlis , could you please rebase with master (also squash 6 commits) branch to see if the the spec:javascript is still failing in GitHub actions
> > 
> > 
> > Hey, I have rebased the branch and fixed one of the failing specs, but the other keeps failing.
> 
> Could you try running them locally to see if this is related to your changes?

I ran the spec locally, and the same spec fails.
This is weird because I cannot see any change I made that would cause the returned status to be 500, but I will keep investigating.

Do you have any idea what could have caused the failing spec?

Oferlis commented:
Changing this PR to WIP since we decided that capabilities will be shown from the physical storage, not from the storage family

MelsHyrule commented:
Manually adding WIP label (for the bot to add it automatically change the title to be [WIP] rather than WIP)

agrare commented on app/helpers/physical_storage_helper/textual_summary.rb:
```diff
@@ -97,4 +97,16 @@ def textual_rack_name
+
+  def parse_capabilities(storage_family_id)
+    capabilities = []
+    if storage_family_id
+      capabilities = PhysicalStorageFamily.find(storage_family_id)&.capabilities
+    end
```
Two things:
1. Prefer setting the physical_storage capabilities to be equal to the physical_storage_family.capabilities in your refresh parser so we only have to hit one location here to show them.  Consider someone using the API where this logic won't exist (and shouldn't have to exist)
2. If you were going to reference the storage_family for some reason you should use the assoication rather than doing a manual query (e.g. `@record.physical_storage_family&.capabilities` don't actually do this here, but if you had to in the future use the assoc not a manual query)

agrare commented on spec/helpers/physical_storage_helper/textual_summary_spec.rb:
```diff
@@ -58,9 +58,9 @@
 
-    it 'shows 8 fields' do
+    it 'shows 9 fields' do
       expect(subject.items).to be_kind_of(Array)
-      expect(subject.items.size).to eq(8)
+      expect(subject.items.size).to eq(9)
```
Not a reflection on your changes at all, but this seems like a pointless test.  Better to check the contents than a whole test checking the number of fields.

jeffibm commented:
Hey @Oferlis , is this PR good to test/merge?

jeffibm commented:
> Hey @Oferlis , is this PR good to test/merge?

Hey, any updates here? or shall we merge it?

Oferlis commented:
Hey, sorry for the late response. 
You may merge it, thanks
